Fix #33 ResultSet.getTimestamp is off by -12 hours between 12:00-13:00

jhannes opened the pull request:
NB: This is a really critical bug! SQLDroid currently reads some timestamps (between 12:00:00-12:59:59) with the wrong value! This is silently wrongful behavior that causes mysterious bugs.

We got this bug reported to us as "some of the times in the database looks like they are wrong." This is a really hard issue to debug, as we had no idea where the error occurred and didn't initially suspect SQLDroid (or any third party library).

donv commented:
Hi @jhannes !

Thanks for pull request!  Could you add a test for this?  Commit rights will be rewarded 😄

jhannes commented:
@donv I've added a test and also extracted the timestamp format as a constant as per discussion with @ergouser.

Unfortunately, I've been unable to find out how to run the tests. So I had to commit it "blindly".

I've added an extra pull request which ports the tests to work under Maven with Robolectric.

PS: On a personal note: We need to catch up soon. :-)

donv commented:
The other PR has been merged.  Is this PR obsolete, or is it now easier to add a proper test?

On catching up:  YES!  Going to any conferences anytime soon?

donv commented:
@jhannes can you rebase this PR so it merges cleanly?  Also, if you could add a section to the README on how to run the tests, that would be great.  I expect it is

    mvn install -B

or some such.

jhannes commented:
My other pull request subsumes this one.

@donv: Going to JavaZone of course. And I am organizing this conference: http://mobileera.rocks/. The CfP deadline is tonight! Hint, hint!
